test/ctia/http/generative/fulltext_search_test.clj

```diff
@@ -106,110 +76,138 @@
                                                            (-> % :title (= "Etiam vel neque bibendum dignissim")))))]
-                           (is (= 3 (count matching)))))}]))
+                           (is (= 3 (count matching)))))}]
 
-(defn test-search-case
-  [app
-   {:keys [query-params
-           bundle-gen
-           check
-           test-description] :as test-case}]
-  (let [services (app/service-graph app)
-        bundle   (gen/generate bundle-gen)
-        ent-keys (->> bundle
-                      keys
-                      (filter (partial contains? bundle-entity-field-names)))]
-    (bundle/import-bundle
-     bundle
-     nil    ;; external-key-prefixes
-     login
-     services)
+   [{:test-description "searching for non-existing fields, returns errors"
+     :query-params     {:query         "*"
+                        :search_fields ["donec_retium_posuere_tellus"
+                                        "proin_neque_massa"]}
+     :bulk-gen         (bulk-gen-for :incidents)
+     :check            (fn [_ _ _ res]
+                         (is (-> res :parsed-body (contains? :errors))))}
+    {:test-description "searching for non-existing values, returns nothing"
+     :query-params     {:query "nullam tempus nulla posuere pellentesque dapibus suscipit ligula"}
+     :bulk-gen         (bulk-gen-for :incidents)
+     :check            (fn [_ _ _ res]
+                         (is (-> res :parsed-body empty?)))}]
+   [(let [bulk (gen/fmap (fn [examples]
+                           (update examples :assets
+                                   (fn [incidents]
+                                     (apply utils/update-items incidents
+                                            [;; update only the first record, leave the rest unchanged
+                                             #(assoc % :short_description "first incident"
+                                                     :title "Lorem Ipsum Test Incident")]))))
+                         (bulk-gen-for :assets))]
+      {:test-description "searching within fields that don't contain the pattern, returns empty col"
+       :query-params     {:query         "first Ipsum"
+                          :search_fields ["description"]}
+       :bulk-gen         bulk
+       :check            (fn [_ _ _ res] (is (-> res :parsed-body empty?)))}
 
+      {:test-description "searching in fields that contain the pattern, yields results"
+       :query-params     {:query         "\"first\" \"Ipsum\""
+                          :search_fields ["short_description" "title"]}
+       :bulk-gen         bulk
+       :check            (fn [_ _ _ unparsed]
+                           (let [res (-> unparsed :parsed-body)]
+                             (is (= 1 (count res)))
+                             (is "first incident" (-> res first :short_description))
+                             (is "Lorem Ipsum Test Incident" (-> res first :title))))})]))
```

Comment:
You tested previous PR here and there is at least one case which is already checked.
Here the task is the default behavior:
``` clojure
{:test-description 'searching in default search fields that contain the pattern, yields results"
       :query-params     {:query         "\"first\" \"Ipsum\""}
       :bulk-gen         bulk
       :check            (fn [_ _ _ unparsed]
                           (let [res (-> unparsed :parsed-body)]
                             (is (= 1 (count res)))
                             (is "first incident" (-> res first :short_description))
                             (is "Lorem Ipsum Test Incident" (-> res first :title))))}
                             
{:test-description 'searching in non default search fields that contain the pattern, shall not return any result"
       ;; green should be the value of :tlp, please do check it
       :query-params     {:query         "green"} 
       :bulk-gen         (assoc bulk
       :check            (fn [_ _ _ unparsed]
                           (let [res (-> unparsed :parsed-body)]
                             (is (empty? res))))}
```

Reply:
You also had issues with nested fields, you need to check this edge case with sightings for instance: `targets.observable.value`